ConcordiumWallet/Views/Stake/Baking/ComissionSettings/BakerCommissionSettingsViewModel.swift

```
@@ -0,0 +1,141 @@
+
+                if let data = self.handler.getNewEntry(BakerComissionData.self) {
+                    self.updateCommisionValues(
+                        baking: data.bakingRewardComission,
+                        transaction: data.transactionComission,
+                        finalization: data.finalizationRewardComission
+                    )
+                    return
+                }
+                if let data = self.handler.getCurrentEntry(BakerComissionData.self) {
+                    self.updateCommisionValues(
+                        baking: data.bakingRewardComission,
+                        transaction: data.transactionComission,
+                        finalization: data.finalizationRewardComission
+                    )
+                    return
+                }
+                self.updateCommisionValues(
+                    baking: response.bakingCommissionRange.max,
+                    transaction: response.transactionCommissionRange.max,
+                    finalization: response.finalizationCommissionRange.max
+                )
+
+                self.commissionRanges = (bakingCommissionRange: response.bakingCommissionRange,
+                               transactionCommissionRange: response.transactionCommissionRange,
+                               finalizationCommissionRange: response.finalizationCommissionRange
+                )
+
+            case let .failure(error):
+                self.error = .networkError(error)
+            }
+        }
+        .store(in: &cancellables)
+    }
+
+    func continueButtonTapped() {
+        do {
+            try validate()
+            handler.add(entry: BakerComissionData(bakingRewardComission: bakingRewardCommission, finalizationRewardComission: finalizationRewardCommission, transactionComission: transactionFeeCommission))
+            didTapContinue()
+        } catch let error {
+            self.error = error as? BakerCommissionSettingError
```

So if the error isn't a `BakerCommissionSettingError` we fail silently?

---

Would not be necessary with the suggestion below.